avm/res/azure-stack-hci/cluster/deployment-settings/main.bicep

```diff
@@ -0,0 +1,213 @@
+param subnetMask string
+
+@description('Required. The default gateway of the Management Network. Exameple: 192.168.0.1.')
+param defaultGateway string
+
+@description('Required. The starting IP address for the Infrastructure Network IP pool. There must be at least 6 IPs between startingIPAddress and endingIPAddress and this pool should be not include the node IPs.')
+param startingIPAddress string
+
+@description('Required. The ending IP address for the Infrastructure Network IP pool. There must be at least 6 IPs between startingIPAddress and endingIPAddress and this pool should be not include the node IPs.')
+param endingIPAddress string
+
+@description('Required. The DNS servers accessible from the Management Network for the HCI cluster.')
+param dnsServers array
+
+@description('Required. An array of Network ATC Network Intent objects that define the Compute, Management, and Storage network configuration for the cluster.')
+param networkIntents array
+
+@description('Required. Specify whether the Storage Network connectivity is switched or switchless.')
+param storageConnectivitySwitchless bool
+
+@description('Required. Enable storage auto IP assignment. This should be true for most deployments except when deploying a three-node switchless cluster, in which case storage IPs should be configured before deployment and this value set to false.')
+param enableStorageAutoIp bool = true
+
+@description('Required. An array of JSON objects that define the storage network configuration for the cluster. Each object should contain the adapterName, VLAN properties, and (optionally) IP configurations.')
+param storageNetworks array
+
+// other cluster configuration parameters
+@description('Required. The name of the Custom Location associated with the Arc Resource Bridge for this cluster. This value should reflect the physical location and identifier of the HCI cluster. Example: cl-hci-den-clu01.')
+param customLocationName string
+
+@description('Required. The name of the storage account to be used as the witness for the HCI Windows Failover Cluster.')
+param clusterWitnessStorageAccountName string
+
+@description('Required. The name of the key vault to be used for storing secrets for the HCI cluster. This currently needs to be unique per HCI cluster.')
+param keyVaultName string
+
+var arcNodeResourceIds = [
+  for (nodeName, index) in clusterNodeNames: resourceId('Microsoft.HybridCompute/machines', nodeName)
+]
+
+var storageNetworkList = [
+  for (storageAdapter, index) in storageNetworks: {
+    name: 'StorageNetwork${index + 1}'
+    networkAdapterName: storageAdapter.adapterName
+    vlanId: storageAdapter.vlan
+    storageAdapterIPInfo: storageAdapter.?storageAdapterIPInfo
+  }
+]
+
+resource cluster 'Microsoft.AzureStackHCI/clusters@2024-04-01' existing = {
+  name: clusterName
+}
+
+resource deploymentSettings 'Microsoft.AzureStackHCI/clusters/deploymentSettings@2024-04-01' = {
+  name: 'default'
```

Comment:
For consistency, I'd recommend to move this value to a parameter with a default value `default`. The perk: Users would actually see this value (e.g., in the readme)

Reply:
The API only accept 'default' as the name - should it still be a parameter? @AlexanderSehr

Reply:
There are quite a few resources that enforce a single value. Another one is for example storage account blob services:
https://github.com/Azure/bicep-registry-modules/blob/6b63b1051ff4e19261a414acc829f6278ac8c3a1/avm/res/storage/storage-account/blob-service/main.bicep#L67-L68

or storage account file services
https://github.com/Azure/bicep-registry-modules/blob/f06337032c4595246698b33c0d63fbb1ef72b80d/avm/res/storage/storage-account/file-service/main.bicep#L9-L10

which already is inconsistent as you can see. But the point of moving the value to anything outside of the deployment block is to make it accessible for any other deployment. Not sure if that's the case for the diagnosticSettings resource, but e.g., in the two cases above, the corresponding variable/parameter (which is **not** set by it's parent module, i.e., the default always applies) is referenced in subsequent resource deployments as part of their own name.

Utlimately it's up to you, but if for nothing else, I'd at least suggest for consistency 😄

Reply:
Since there is no valid value for this name except default according to the API, I don't see value in declaring it in a parameter or variable

Reply:
Like described above. The only perk is that users are actually told about the value. If that's worth anything is of course a different conversation. I'll leave that to you. I don't think there's a spec for it